Felienne/hedy
# [FEATURE] Enables teachers to directly add adventures to classes

**Description**
In this PR we add a nice feature to how class customizations work. From now on teachers are able to directly store their custom adventure in their classes. This seems like a simply task, but due to our data structuring was quite complex to code. We have to retrieve all classes as well as all customizations on each adventure call; which is quite an expensive operation. However, we expect that this has not much influence on performance as this page will not be loaded often.

Adventures can be directly checked or unchecked on classes to automatically add them to the customizations. If customizations do not exist (yet) they are created with only the teacher adventures with a value. This is automatically handled correctly by the back-end when the end user retrieves the coding page. Extra tests are written to verify that everything works as expected.

**Fixes**
This PR fixes #3083.

**How to test**
Make sure you are a teacher and create (or edit an already existing) adventure. Notify the new buttons on the bottom of the page, representing all your current classes. Verify that you can select and de-select these classes which will automatically customize the classes with your current adventure.
